src/search.cpp

```diff
@@ -363,7 +363,7 @@ void Thread::search() {
   // Iterative deepening loop until requested to stop or the target depth is reached
   while (   (rootDepth += ONE_PLY) < DEPTH_MAX
          && !Threads.stop
-         && !(Limits.depth && mainThread && rootDepth / ONE_PLY > Limits.depth))
+         && !(Limits.depth && mainThread && Cluster::is_root() && rootDepth / ONE_PLY > Limits.depth))
```

Comment:
This disables skipping search depth on all non-root nodes, is it intended?

Reply:
Sorry, it's a not condition, so this enables skipping search depth on all non-root nodes, including `mainThread`.
This means that depth all non-root nodes can go very wild(snowball effect since we basically don't have a `mainThread` there)...